common/network-common/src/main/java/org/apache/spark/network/server/TransportServer.java

```diff
@@ -105,11 +104,12 @@ private void init(String hostToBind, int portToBind) {
     EventLoopGroup workerGroup =  NettyUtils.createEventLoop(ioMode, conf.serverThreads(),
       conf.getModuleName() + "-server");
 
+    boolean isNotWindows = !System.getProperty("os.name").regionMatches(true, 0, "Windows", 0, 7);
```

Comment:
I wonder why we just use `startsWith` like `isOsNameMatch` did?

Reply:
You are correct, @viirya . However, in JVM implementation, "os.name" always exists. I'm not aware of any JVM implementations who `os.name` value is NULL. I don't think we need to check `null` for `os.name` here because this PR considers a specialized case in Spark runtime environment while `commons-lang3` are trying to the generalized versions.

Reply:
How about length check?